src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,8 +102,22 @@ public VSSolutionManager()
                 ErrorHandler.ThrowOnFailure(hr);
             }
 
-            UserAgent.SetUserAgentString(
-                new UserAgentStringBuilder().WithVisualStudioSKU(VSVersionHelper.GetFullVsVersionString()));
+            // Allow this constructor to be invoked from either the UI or a worker thread
+            if (ThreadHelper.CheckAccess())
```

Comment:
Got the same feedback from Yishai. I was being overly careful it seems 😄. Will fix.